backend/src/main/java/com/bakdata/conquery/apiv1/QueryProcessor.java

```diff
@@ -400,42 +436,56 @@ public ExternalUploadResult uploadEntities(Subject subject, Dataset dataset, Ext
 
-		final ConceptQuery entitySelectQuery =
-				new ConceptQuery(new CQDateRestriction(Objects.requireNonNullElse(dateRange, Range.all()), new CQExternal(List.of(idKind), new String[][]{{"HEAD"}, {entity}}, false)));
+		final ConceptQuery entitySelectQuery = new ConceptQuery(new CQExternal(List.of(idKind), new String[][]{{"HEAD"}, {entity}}, true));
 
 		final TableExportQuery exportQuery = new TableExportQuery(entitySelectQuery);
-		exportQuery.setTables(
-				sources.stream()
-					   .map(source -> {
-						   final CQConcept cqConcept = new CQConcept();
-						   cqConcept.setElements(List.of(source.getConcept()));
-						   final CQTable cqTable = new CQTable();
-						   cqTable.setConcept(cqConcept);
-						   cqTable.setConnector(source);
-						   cqConcept.setTables(List.of(cqTable));
+		exportQuery.setDateRange(dateRange);
+
+		exportQuery.setTables(sources.stream().map(CQConcept::forConnector).collect(Collectors.toList()));
 
-						   return cqConcept;
-					   })
-					   .collect(Collectors.toList())
-		);
+		final ManagedQuery execution = (ManagedQuery) postQuery(dataset, exportQuery, subject, true);
 
-		final ManagedExecution<?> execution = postQuery(dataset, exportQuery, subject, true);
+		final List<Select> infoCardSelects = getInfoCardSelects(config.getPreview(), dataset, datasetRegistry);
+
+
+		final AbsoluteFormQuery infoCardQuery =
+				new AbsoluteFormQuery(entitySelectQuery, dateRange,
+									  ArrayConceptQuery.createFromFeatures(
+											  infoCardSelects.stream()
+															 .map(CQConcept::forSelect)
+															 .collect(Collectors.toList())),
+									  List.of(ExportForm.ResolutionAndAlignment.of(Resolution.COMPLETE, Alignment.NO_ALIGN))
+				);
 
+		final ManagedQuery infoCardExecution = (ManagedQuery) postQuery(dataset, infoCardQuery, subject, true);
 
-		// collect id immediately so it does not get sucked into closure
-		final ManagedExecutionId id = execution.getId();
 
-		while (execution.awaitDone(10, TimeUnit.SECONDS) == ExecutionState.RUNNING) {
-			log.trace("Still waiting for {}", id);
+		if (execution.awaitDone(10, TimeUnit.SECONDS) == ExecutionState.RUNNING) {
+			log.warn("Still waiting for {} after 10Seconds.", execution.getId());
+			throw new ConqueryError.ExecutionProcessingTimeoutError();
+		}
+
+		if (infoCardExecution.awaitDone(10, TimeUnit.SECONDS) == ExecutionState.RUNNING) {
+			log.warn("Still waiting for {} after 10Seconds aborting", infoCardExecution.getId());
+			throw new ConqueryError.ExecutionProcessingTimeoutError();
 		}
 
 		if (execution.getState() == ExecutionState.FAILED) {
 			throw ConqueryError.ContextError.fromErrorInfo(execution.getError());
 		}
 
+		if (infoCardExecution.getState() == ExecutionState.FAILED) {
+			throw ConqueryError.ContextError.fromErrorInfo(infoCardExecution.getError());
+		}
```

Comment:
infoCardExecution und execution IFs sind hier etwas in einander verflochten/abwechselnd

Reply:
Ich weiß nicht was du damit meinst.

Reply:
execution in Z463
infoCardExecution in Z468
execution in Z473
infoCardExecution in Z477

Warum nicht erstmal execution das Fehlerhandling machen und dann bei infoCardExecution ?
Oder hast du es implizit gewollt, dass infoCardExecution biszu 20 Sekunden Zeit hat und execution nur 10 Sekunden?

Reply:
Ne eigentlich sind 10s viel zu viel, es würden 1s reichen bei infocard